src/main/java/org/reaktivity/nukleus/tcp/internal/stream/WriteStream.java

```diff
@@ -276,6 +277,10 @@ private void doCleanup()
             try
             {
                 channel.shutdownOutput();
+                if(channel.socket().isInputShutdown())
+                {
+                    CloseHelper.quietClose(channel);
+                }
```

Comment:
Hoist try catch into `shutdownOutput()` method, and new code into `closeIfInputShutdown()` method, leaving:
```
if (channel.isConnected())
{
    shutdownOutput();
    closeIfInputShutdown();
}
```